proto/xudrpc.proto

```diff
@@ -366,6 +373,57 @@ message ListPeersResponse {
   repeated Peer peers = 1 [json_name = "peers"];
 }
 
+message ListSwapDealsRequest {
+  bool all = 1 [json_name = "all"];
+}
+message ListSwapDealsResponse {
+  repeated Swap swaps = 1 [json_name = "swaps"];
+}
+
+message Swap {
+  SwapRole role = 1 [json_name = "role"];
```

Comment:
Adding every field from the database seems like overkill from an rpc perspective. I think it'd make more sense to have a `oneof` for either a `SwapSuccess` or `SwapFailure` (there are examples of `oneof` elsewhere in the proto file) plus `create_time`.  I'm thinking we may want to add more fields to our `SwapFailure` message (like the failure reason), but it's ok for now. Let me know if you have any questions.

Reply:
@sangaman Should the `SwapSuccess` be the same as [this type](https://github.com/ExchangeUnion/xud/blob/d61b07f7c3b2678f08e0dde54e66ef0095853e5a/lib/swaps/types.ts#L59)

Reply:
Yes, although the `SwapDealInstance` type from the database should have all the fields you need to create a `SwapSuccess` gRPC message. You should be able to reuse the `createSwapSuccess` and `createSwapFailure` methods in GrpcService.ts.